lib/spack/spack/build_environment.py

```diff
@@ -1072,6 +1076,15 @@ def set_all_package_py_globals(self):
                 pkg.setup_dependent_package(dependent_module, spec)
                 dependent_module.propagate_changes_to_mro()
 
+        pkg = self.specs[0].package
```

Comment:
What if there is no spec? :slightly_smiling_face: This line added a noisy warning on `develop`:
```console
$ spack env activate --temp
==> Warning: could not unload runtime environment due to IndexError: tuple index out of range
==> Warning: could not load runtime environment due to IndexError: tuple index out of range
==> Created and activated temporary environment in /tmp/spack-xgck4w5c
```
Can you submit a fix for this?